core/src/main/java/com/linecorp/armeria/server/Server.java

```diff
@@ -691,6 +685,39 @@ protected void closeFailed(Throwable cause) {
         private void logStopFailure(Throwable cause) {
             logger.warn("Failed to stop a server: {}", cause.getMessage(), cause);
         }
+
+        private void shutdownExecutor(Iterable<ScheduledExecutorService> executors) {
+            for (ScheduledExecutorService executor : executors) {
+                if (executor instanceof UnstoppableScheduledExecutorService) {
+                    executor = ((UnstoppableScheduledExecutorService) executor).getExecutorService();
+                }
+
+                executor.shutdown();
+            }
+
+            boolean interrupted = false;
+            for (ScheduledExecutorService executor : executors) {
+                if (executor instanceof UnstoppableScheduledExecutorService) {
+                    executor = ((UnstoppableScheduledExecutorService) executor).getExecutorService();
+                }
+
+                try {
+                    while (!executor.isTerminated()) {
+                        try {
+                            executor.awaitTermination(1, TimeUnit.DAYS);
+                        } catch (InterruptedException ignore) {
+                            interrupted = true;
+                        }
+                    }
+                } catch (Exception e) {
+                    logger.warn("Failed to shutdown the blockingTaskExecutor: {}", executor, e);
+                } finally {
+                    if (interrupted) {
+                        Thread.currentThread().interrupt();
+                    }
+                }
+            }
```

Comment:
I think we need to propagate the event outside of the for loop so that the next `executor` in `executors` isn't interrupted right away when `executor.awaitTermination` is called.
```suggestion
            for (ScheduledExecutorService executor : executors) {
                if (executor instanceof UnstoppableScheduledExecutorService) {
                    executor = ((UnstoppableScheduledExecutorService) executor).getExecutorService();
                }

                try {
                    while (!executor.isTerminated()) {
                        try {
                            executor.awaitTermination(1, TimeUnit.DAYS);
                        } catch (InterruptedException ignore) {
                            interrupted = true;
                        }
                    }
                } catch (Exception e) {
                    logger.warn("Failed to shutdown the blockingTaskExecutor: {}", executor, e);
                }
            }
            if (interrupted) {
                Thread.currentThread().interrupt();
            }
```

Reply:
Yes!! I miss that